### ISSUE TITLE
Devices can be controlled from Homebridge but do not respond in Apple Homekit.

### ISSUE THREAD

**Issue by WimBre** [homebridge/homebridge#3314]
### Current Situation

Hi,

I have a number of devices that I can control with Homekit:

Philips Hue lights (via Philips Hub)
Xiaomi motion sensors (via Xiaomi hub)
Xiaomi water sensor (via Xiaomi hub)
Homepods
Eneco Toon
Outdoor lighting (via Sonos/Ewelink)
Bosch dishwasher.
The strange thing is that I can no longer access devices 5 to 7 via Homekit (status: updating). The outdoor lighting cannot be turned on/off, Eneco does not show the degrees.

However, I can control them through Homebridge and I can see the temperature of the Toon nicely and turn the outdoor lighting on/off.

I removed them  from Homekit and tried to add a new Bridge to Homekit but without suc6.

Does anyone have experience with this?

### Logs

```markdown
[01/02/2023, 17:04:13] [HB Supervisor] Starting Homebridge with extra flags: -I
[01/02/2023, 17:04:13] [HB Supervisor] Started Homebridge v1.6.0 with PID: 11444
[01/02/2023, 17:04:14] Loaded config.json with 0 accessories and 4 platforms.
[01/02/2023, 17:04:15] Loaded 3 cached accessories from cachedAccessories.
[01/02/2023, 17:04:15] ---
[01/02/2023, 17:04:17] Loaded plugin: homebridge-config-ui-x@4.50.2
[01/02/2023, 17:04:17] Registering platform 'homebridge-config-ui-x.config'
[01/02/2023, 17:04:17] ---
[01/02/2023, 17:04:17] Loaded plugin: homebridge-ewelink@10.4.0
[01/02/2023, 17:04:17] Registering platform 'homebridge-ewelink.eWeLink'
[01/02/2023, 17:04:17] ---
[01/02/2023, 17:04:18] Loaded plugin: homebridge-homeconnect@0.27.0
[01/02/2023, 17:04:18] Registering platform 'homebridge-homeconnect.HomeConnect'
[01/02/2023, 17:04:18] ---
[01/02/2023, 17:04:20] Loaded plugin: homebridge-ring@11.7.1
[01/02/2023, 17:04:20] Registering platform 'homebridge-ring.Ring'
[01/02/2023, 17:04:20] ---
[01/02/2023, 17:04:20] Loaded plugin: homebridge-toon@2.0.5
[01/02/2023, 17:04:20] Registering platform 'homebridge-toon.Toon'
[01/02/2023, 17:04:20] ---
[01/02/2023, 17:04:20] Loading 4 platforms...
[01/02/2023, 17:04:20] [<TOON>] Initializing Toon platform...
[01/02/2023, 17:04:20] [eWeLink] Initializing eWeLink platform...
[01/02/2023, 17:04:20] [eWeLink] Initialising plugin v10.4.0 | System linux | Node v18.13.0 | HB v1.6.0 | HAPNodeJS v0.11.0...
[01/02/2023, 17:04:20] [eWeLink] Config entry [groups] is unused and can be removed.
[01/02/2023, 17:04:20] [Ring] Initializing Ring platform...
[01/02/2023, 17:04:20] [<TOON>] Getting agreement...
[01/02/2023, 17:04:20] [Ring] Configuring cached accessory 3ae38777-cf7d-4fde-b077-802ccf074c18 Camera voordeur 
[01/02/2023, 17:04:20] [<TOON>] Toon Initialized, it may take a few minutes before any data will be visible to HomeKit.
[01/02/2023, 17:04:20] [eWeLink] Plugin initialised. Setting up accessories....
Setup Payload:
X-HM://0024QEV4UA427
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 763-90-174 │     
    └────────────┘     
                       
[01/02/2023, 17:04:21] Homebridge v1.6.0 (HAP v0.11.0) (Homebridge 575E) is running on port 51610.
[01/02/2023, 17:04:21] [<TOON>] Currently selected agreementIndex: 0
[01/02/2023, 17:04:22] [<TOON>] Current Temperature:  undefined
[01/02/2023, 17:04:22] [homebridge-toon] This plugin generated a warning from the characteristic 'Current Temperature': characteristic value expected valid finite number and received "undefined" (undefined). See https://homebridge.io/w/JtMGR for more info.
[01/02/2023, 17:04:22] [<TOON>] Current Target Temperature:  undefined
[01/02/2023, 17:04:22] [homebridge-toon] This plugin generated a warning from the characteristic 'Target Temperature': characteristic value expected valid finite number and received "undefined" (undefined). See https://homebridge.io/w/JtMGR for more info.
[01/02/2023, 17:04:22] [Ring] Found the following locations:
[01/02/2023, 17:04:22] [Ring]   locationId: 6e96e273-1fc4-494a-86f7-14ebc08102e2 - xxxxxxxxxx
[01/02/2023, 17:04:22] [eWeLink] Fetching devices from eWeLink home/id [Mijn huis] [5fd479ac8e79e30008ede16a].
[01/02/2023, 17:04:23] [Ring] Configuring 1 cameras and 1 devices for location "xxxxxxxxxx" - locationId: 6e96e273-1fc4-494a-86f7-14ebc08102e2
[01/02/2023, 17:04:26] [eWeLink] LAN monitoring started.
[01/02/2023, 17:04:26] [eWeLink] [Buitenverlichting voordeur] initialising with options {"inUsePowerThreshold":0,"logging":"standard","showAs":"default"}.
[01/02/2023, 17:04:26] [eWeLink] [Buitenverlichting voordeur] initialised and found locally with IP [192.168.178.211] [1:BASICR2].
[01/02/2023, 17:04:28] [eWeLink] ✓ Setup complete. Have time to give this plugin a review? https://bit.ly/hb-ewelink-review
[01/02/2023, 17:10:41] [Homebridge UI] Starting terminal session
[01/02/2023, 17:11:05] [Homebridge UI] Terminal session ended.
[01/02/2023, 17:11:16] [Homebridge UI] Starting terminal session
[01/02/2023, 17:11:41] [Homebridge UI] Terminal session ended.
[01/02/2023, 17:31:42] [Ring] Camera voordeur  Detected Motion. Loading snapshot before sending event to HomeKit
[01/02/2023, 17:36:47] [Homebridge UI] Starting terminal session
[01/02/2023, 17:37:51] [Homebridge UI] Terminal session ended.
[01/02/2023, 17:37:53] [Homebridge UI] Starting terminal session
[01/02/2023, 17:38:04] [Homebridge UI] Terminal session ended.
[01/02/2023, 17:38:15] [Homebridge UI] Starting terminal session
[01/02/2023, 17:40:04] [Homebridge UI] Terminal session ended.
```

### Configuration

```json
"bridge": {
        "name": "Homebridge 575E",
        "username": "xxxxxxxxxxxxxxxxxxxxx",
        "port": 51610,
        "pin": "763-90-174"
    },
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        },
        {
            "platform": "Toon",
            "name": "<TOON>",
            "username": "xxxxxxxxx",
            "password": "xxxxxxxxxxxx",
            "consumerKey": "xxxxxxxxxxxxxxxxxxxxxxxxxxx",
            "consumerSecret": "xxxxxxxxxxxxxx",
            "apiToken": "xxxxxxxxxxxxxxxxxxxxxxxxxxxx"
        },
        {
            "name": "eWeLink",
            "countryCode": "31",
            "username": "xxxxxxxxxxxxxxxxxxxx",
            "password": "xxxxxxxxxxxxxxxxxxxx",
            "groups": [
                {
                    "type": "blind"
                }
            ],
            "platform": "eWeLink"
        },
        {
            "refreshToken": "xxxxxxxxxxxxxxxxxxxxx",
            "platform": "Ring"
        }
    ]
}
```

### Environment

- OS: Linux homebridge 5.10.103-v7l+
- Node: v18.13.0
- npm: npm@8.19.3 
- Homebridge: 1.6.0

### Process Supervisor

hb-service

**Comment by kevinjalbert** [comment]
I have this same issue, but with TPlink and Yolink plugins. No matter what I do, I cannot get them to work correctly anymore. It just shows up in HomeKit (initially with the right state), but is unresponsive.

I can still use the homebridge interface tho to interact with the devices. So this seems like a Homebridge <-> Homekit issue?

**Comment by kevinjalbert** [comment]
https://github.com/homebridge/homebridge/issues/3315#issuecomment-1413728996 Looks like it might have the same issue.

> I removed all bridges from home and setup a new VM setup one plugin and added it to home. Devices are listed with correct state and if i try to control them -> no response

**Comment by kevinjalbert** [comment]
I downgraded to 1.5.0 and redid everything and this time it worked (I used the ciao mDNS Advertiser as well).

**Comment by bennettp123** [comment]
> Chiming in with a "me too"/"+1". Definitely seems like an issue between HomeKit and Homebridge. Just started happening yesterday, but a lot more severe today.
> 
> I can usually get it to start working again by restarting Homebridge, but it usually stops after a minute or two.
> 
> I can usually see bridges & child bridges listed in [mDNS](https://apps.apple.com/au/app/discovery-dns-sd-browser/id1381004916?mt=12), but starting today, they disappear a few seconds after homebridge starts.
> 
> Switching from Ciao to Bonjour-HAP seems like it's helped, though it's only been a couple of minutes—I'll edit this post if it breaks again.
> 
> Edit: I've switched back to ciao to reproduce the issue in ciao with debug logs enabled (`DEBUG=ciao:*`)
> 
> Debug logs: https://gist.github.com/bennettp123/7088a80e550db7e9bbc24e7a1f2d874d
> 
> It sends out a few advertisements, then just locks up, doesn't send anything at all. 🤷🏻
> 
> Bonjour-HAP still working fine 👍🏻
> 
> Edit 2: I just realised avahi is running inside the official docker image—I should probably be using that instead of Ciao/Bonjour-HAP. I assumed (out of habit) that dbus, and therefore avahi, would be unavailable inside the docker image. 🙃

**Comment by bimusiek** [comment]
I have exactly the same problem, neither avahi nor bonjour-hap works for me.

**Comment by bennettp123** [comment]
Another update: a few days ago, Bonjour-HAP stopped working too (previously worked fine).

Apologies but after some more troubleshooting, I think my issue is not 100% related to this thread. It seems libvirt doesn't enable multicast by default, which breaks mDNS. More info: https://superuser.com/questions/944678/how-to-configure-macvtap-to-let-it-pass-multicast-packet-correctly

---
symptoms:

* when the devices are offline in the Home app, I also cannot see homebridge (or any child bridges) advertised in mDNS (using Discovery app—DNS-SD browser)—this indicates there's an issue with my network, not homebridge itself 
* when I restart homebridge, _all_ the child bridges appear! (under _hap._tcp.) — but after a few seconds, they disappear again 😢 
* others in this thread report success after downgrading Homebridge from 1.6.0 to 1.5.x—this makes sense because 1.6.0 includes mDNS changes. Unfortunately, this doesn't fix my problems.
* I also tried switching to Home Assistant—and it _also_ has the same issue! The bridge appears in mDNS, then disappears after a few seconds. 

Anyway, these symptoms suggest it's an issue with multicast on my network, not homebridge at all. And once the multicast issues were fixed, Homebridge again seems to be working fine.
